tests/functional/keep-going.nix

```diff
@@ -0,0 +1,33 @@
+with import ./config.nix;
+
+rec {
+
+  # Hack to get the scheduler to do what we want: The `good` derivation can
+  # only be built after `delay_good` (which takes a long time to build) while
+  # the others don't have any dependency.
+  # This means that if we build this with parallelism (`-j2`), then we can be
+  # reasonably sure that the failing derivations will be scheduled _before_ the
+  # `good` one (and so we can check that `--keep-going` works fine)
```

Comment:
This could be synchronized by waiting for a file to appear in a mutable directory in `extra-sandbox-paths`. The test would have to monitor the CLI's log before creating it.
(doing it in the `failing` derivation would only decrease the length of the race condition, but not make it reliable)

Something along the lines of

```bash
nix build --extra-sandbox-paths "$TEST_ROOT/sync" 2>&1 \
  | while read ln; do
    echo "nix build: $ln"
    if echo "$ln" | grepQuiet ...; then
      touch "$TEST_ROOT/sync/failing-drv-failed"
    fi
  done
```